state/state_test.go

```diff
@@ -0,0 +1,68 @@
+package state
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/tendermint/basecoin/types"
+	eyes "github.com/tendermint/merkleeyes/client"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestState(t *testing.T) {
+
+	s := NewState(types.NewMemKVStore())
+
+	s.SetChainID("testchain")
+	assert.True(t, s.GetChainID() == "testchain", "ChainID is improperly stored")
+
+	setRecords := func(kv types.KVStore) {
+		kv.Set([]byte("foo"), []byte("snake"))
+		kv.Set([]byte("bar"), []byte("mouse"))
+	}
+
+	setRecords(s)
+	assert.True(t, bytes.Equal(s.Get([]byte("foo")), []byte("snake")), "state doesn't retrieve after Set")
```

Comment:
would love to see a more table driven approach for a lot of these tests: https://dave.cheney.net/2013/06/09/writing-table-driven-tests-in-go